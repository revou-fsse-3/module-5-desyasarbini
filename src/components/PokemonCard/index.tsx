import Link from "next/link"

interface Props {
    name: string
}

const PokemonCard = ({name}: Props) => {
    return(
        <Link 
            href={name}
            className="group rounded-full border-2 border-gray-200 m-3 px-5 py-4 text-center transition-colors bg-amber-200 hover:border-gray-300 hover:bg-amber-300"
            key={name + 'card'}
        >
            <h2 className={`text-2xl- front-semibold text-red-800 hover:text-zinc-100`}>
                {name.charAt(0).toUpperCase()+name.slice(1)}
            </h2>
        </Link>
    )
}
export default PokemonCard