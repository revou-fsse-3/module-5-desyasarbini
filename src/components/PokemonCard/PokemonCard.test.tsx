import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import PokemonCard from '.';

describe('PokemonCard component', () => {
    const mockProps = {
        name: 'pikachu',
    };

    test('renders PokemonCard correctly', () => {
        const { asFragment } = render(<PokemonCard {...mockProps} />);
        
        // asFragment() untuk mengambil snapshot dari struktur DOM yang dihasilkan oleh komponen
        expect(asFragment()).toMatchSnapshot();
    });
});
