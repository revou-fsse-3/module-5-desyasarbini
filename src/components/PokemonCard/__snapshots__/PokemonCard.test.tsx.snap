// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonCard component renders PokemonCard correctly 1`] = `
<DocumentFragment>
  <a
    class="group rounded-full border-2 border-gray-200 m-3 px-5 py-4 text-center transition-colors bg-amber-200 hover:border-gray-300 hover:bg-amber-300"
    href="pikachu"
  >
    <h2
      class="text-2xl- front-semibold text-red-800 hover:text-zinc-100"
    >
      Pikachu
    </h2>
  </a>
</DocumentFragment>
`;
